api/rds.proto

```diff
@@ -585,6 +588,22 @@ message RedirectAction {
   RedirectResponseCode response_code = 3 [(validate.rules).enum.defined_only = true];
 }
 
+message DirectResponseAction {
+  // Specifies the HTTP response status to be returned.
+  google.protobuf.UInt32Value status = 1 [(validate.rules).message.required = true];
+
+  // Specifies the human-readable status to be returned if
+  // the response is encoded as HTTP/1.x.
+  string reason = 2;
```

Comment:
Practically it's not going to be possible to actually use this. The HTTP/1.1 codec will select the reason based on the code for you and you can't set it. I would drop this.